management/networksecuritygroup/client.go

```diff
@@ -0,0 +1,235 @@
+)
+
+const (
+	createSecurityGroupURL           = "services/networking/networksecuritygroups"
+	deleteSecurityGroupURL           = "services/networking/networksecuritygroups/%s"
+	getSecurityGroupURL              = "services/networking/networksecuritygroups/%s"
+	listSecurityGroupsURL            = "services/networking/networksecuritygroups"
+	addSecurityGroupToSubnetURL      = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups"
+	getSecurityGroupForSubnetURL     = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups"
+	removeSecurityGroupFromSubnetURL = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups/%s"
+	setSecurityGroupRuleURL          = "services/networksecuritygroups/%s/rules/%s"
+	deleteSecurityGroupRuleURL       = "services/networksecuritygroups/%s/rules/%s"
+
+	errParamNotSpecified = "Parameter %s is not specified."
+)
+
+// NewClient is used to instantiate a new SecurityGroupClient from an Azure client
+func NewClient(client management.Client) *SecurityGroupClient {
+	return &SecurityGroupClient{client: client}
+}
+
+// CreateNetworkSecurityGroup creates a new network security group within
+// the context of the specified subscription
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913818.aspx
+func (sg *SecurityGroupClient) CreateNetworkSecurityGroup(
+	name string,
+	label string,
+	location string) (management.OperationId, error) {
+	if name == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "name")
+	}
+	if location == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "location")
+	}
+
+	req := SecurityGroup{
+		Name:     name,
+		Location: location,
+	}
+
+	if label != "" {
+		req.Label = label
+	}
+
+	data, err := xml.Marshal(req)
+	if err != nil {
+		return "", err
+	}
+
+	requestURL := fmt.Sprintf(createSecurityGroupURL)
+	return sg.client.SendAzurePostRequest(requestURL, data)
+}
+
+// DeleteNetworkSecurityGroup deletes the specified network security group from the subscription
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913825.aspx
+func (sg *SecurityGroupClient) DeleteNetworkSecurityGroup(
+	name string) (management.OperationId, error) {
+	if name == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "name")
+	}
+
+	requestURL := fmt.Sprintf(deleteSecurityGroupURL, name)
+	return sg.client.SendAzureDeleteRequest(requestURL)
+}
+
+// GetNetworkSecurityGroup returns information about the specified network security group
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913821.aspx
+func (sg *SecurityGroupClient) GetNetworkSecurityGroup(name string) (*SecurityGroup, error) {
+	if name == "" {
+		return nil, fmt.Errorf(errParamNotSpecified, "name")
+	}
+
+	requestURL := fmt.Sprintf(getSecurityGroupURL, name)
+	response, err := sg.client.SendAzureGetRequest(requestURL)
+	if err != nil {
+		return nil, err
+	}
+
+	var securityGroup SecurityGroup
+	if err := xml.Unmarshal(response, &securityGroup); err != nil {
+		return nil, err
+	}
+
+	return &securityGroup, nil
```

Comment:
you can rewrite this as:

```go
err = xml.Unmarshal(response, &securityGroup);
return securityGroup, err 
```

gives the same impact really once you change this method to not to return pointer.